stbt-batch.d/run.py

```diff
@@ -152,6 +150,119 @@ def on_term(_signo, _frame):
         return 1
 
 
+GitInfo = namedtuple('GitInfo', 'commit commit_sha git_dir')
+
+
+def read_git_info(testdir):
+    try:
+        def git(*cmd):
+            return subprocess.check_output(('git',) + cmd, cwd=testdir).strip()
+        return GitInfo(
+            git('describe', '--always', '--dirty'),
+            git('rev-parse', 'HEAD'),
+            git('rev-parse', '--show-toplevel'))
+    except subprocess.CalledProcessError:
+        return None
+    except OSError as e:
+        # ENOENT means that git is not in $PATH
+        if e.errno != errno.ENOENT:
+            raise
+
+
+def make_rundir(outputdir, tag):
+    for n in range(2):
+        rundir = datetime.datetime.now().strftime("%Y-%m-%d_%H.%M.%S") + tag
+        try:
+            os.mkdir(os.path.join(outputdir, rundir))
+            return rundir
+        except OSError as e:
+            if e.errno == errno.EEXIST and n == 0:
+                # Avoid directory name clashes if the previous test took <1s to
+                # run
+                time.sleep(1)
+            else:
+                raise
+
+
+def fill_in_data_files(rundir, test_name, test_args, git_info, tag):
+    def write_file(name, data):
+        with open(os.path.join(rundir, name), 'w') as f:
+            f.write(data)
+
+    if git_info:
+        write_file("git-commit", git_info.commit)
+        write_file("git-commit-sha", git_info.commit_sha)
+        write_file("test-name", os.path.relpath(test_name, git_info.git_dir))
+    else:
+        write_file("test-name", test_name)
```

Comment:
Also a regression here: If there's no `.git` then we used to write the absolute path to the file.